src/commcare_cloud/commands/terraform/templates/terraform.tf.j2

```diff
@@ -7,20 +7,23 @@ terraform {
   }
   {#- guaranteed to be a valid string; see comment in terraform_version_range_string definition #}
   required_version = "{{ terraform_version_range_string }}"
-}
-
-provider "aws" {
-  version = "~> 4.8.0"
-  region  = {{ region|tojson }}
-}
 
-provider "archive" {
-  version = "~> 2.2.0"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.8.0"
+    }
+    archive = {
+      source = "hashicorp/archive"
+      version = "~> 2.2.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.1.2"
+    }
+  }
 }
 
-provider "random" {
-  version = "~> 3.1.2"
-}
```

Comment:
Are the changes in this file backwards-compatible? I.e. do they work equally well on 0.14 and 1.2?

Reply:
Yes, these changes are backward compatible with 0.14 and 1.2, as these changes were made after moving to 0.14. Earlier these changes were not made to be backwards compatible with 0.13 and left as warning like **Version constraints inside provider configuration blocks are deprecated.**